strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/io/ArtifactInputStreamTest.java

```diff
@@ -0,0 +1,69 @@
+package org.carlspring.strongbox.io;
+
+/**
+ * Created by dinesh on 1/2/18.
+ */
+
+import java.io.*;
+import java.security.NoSuchAlgorithmException;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertTrue;
+
+public class ArtifactInputStreamTest
+{
+
+
+    ArtifactInputStream artifactInputStream;
+    InputStream inputStream;
+    String pathTxt, pathSh;
+
+    @Before
+    public void setUp()
+
+    {
+        pathTxt = "src/test/resources/artifacts/testTextFile.txt";
+        pathSh = "src/test/resources/artifacts/project-1.4.1.tar.gz";
+
+    }
+
+
+    @Test
+    public void testFileExtensionEqualsTxt()
+            throws FileNotFoundException, NoSuchAlgorithmException
+    {
+        try
+        {
+            inputStream = new BufferedInputStream(new FileInputStream(pathTxt.trim()));
+            artifactInputStream = new ArtifactInputStream(null, inputStream)
+            {
+            };
+            assertTrue(artifactInputStream.getFileExtension(IOUtils.toByteArray(inputStream)).equalsIgnoreCase(".txt"));
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testFileExtensionEqualsGz()
+            throws FileNotFoundException, NoSuchAlgorithmException
+    {
+        try
+        {
+            inputStream = new BufferedInputStream(new FileInputStream(pathSh.trim()));
+            artifactInputStream = new ArtifactInputStream(null, inputStream)
+            {
+            };
+            assertTrue(artifactInputStream.getFileExtension(IOUtils.toByteArray(inputStream)).equalsIgnoreCase(".gz"));
```

Comment:
This is incorrect. The extension here should actually be `tar.gz` (and with no dot).

Reply:
hmm.. I will check if Tika has another method. Currently the detect method returns .gz as file type.